File: backend/app/models/property.py
from datetime import datetime
from sqlalchemy import Column, Integer, String, Float, Boolean, DateTime, Text, ForeignKey
from sqlalchemy.orm import relationship
from ..database import Base


class Property(Base):
    __tablename__ = "properties"

    id = Column(Integer, primary_key=True, index=True)
    address = Column(String(500), nullable=False, index=True)
    house_type = Column(String(50))
    area = Column(Float)
    floor = Column(String(50))
    decoration_status = Column(String(50))
    photo_urls = Column(Text, default="")
    is_rented = Column(Boolean, default=False)
    note = Column(Text, default="")
    
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

    contracts = relationship("Contract", back_populates="property")
    repairs = relationship("Repair", back_populates="property")

File: frontend/src/lib/api/property.ts
import { api } from './client';

export interface Property {
  id: number;
  address: string;
  house_type?: string;
  area?: number;
  floor?: string;
  decoration_status?: string;
  photo_urls?: string;
  is_rented: boolean;
  note?: string;
  created_at: string;
  updated_at: string;
}

export interface PropertyCreate {
  address: string;
  house_type?: string;
  area?: number;
  floor?: string;
  decoration_status?: string;
  photo_urls?: string;
  is_rented?: boolean;
  note?: string;
}

export const propertyApi = {
  list: (params?: { is_rented?: boolean }) => {
    const qs = params?.is_rented !== undefined ? `?is_rented=${params.is_rented}` : '';
    return api.get<Property[]>(`/properties${qs}`);
  },
  get: (id: number) => api.get<Property>(`/properties/${id}`),
  create: (data: PropertyCreate) => api.post<Property>('/properties', data),
  update: (id: number, data: Partial<PropertyCreate>) =>
    api.put<Property>(`/properties/${id}`, data),
  remove: (id: number) => api.del(`/properties/${id}`),
};
